### PR TITLE
Raincatcher 406

### PR DESCRIPTION
**Pull request by JameelB**
## Issue
Raincatcher-user should validate requests coming from the client app to the cloud app using session token. 

## Solution
Created an express middleware to perform session validation. 
Creates a subscriber to this session validation which calls the mbaas validate session endpoint. 

## JIRA Ticket
https://issues.jboss.org/browse/RAINCATCH-406

### CODE REVIEW

File: lib/middleware/validateSession.js [hunk 1]
```diff
@@ -10,14 +10,23 @@ var verifySession = require('./verifySession');
  * @param {Object} mbaasApi object used to access sessions in cache
  * @returns {Function} validateSessionMiddleware function to validate session
  */
-module.exports = function validateSession(mediator, mbaasApi) {
+module.exports = function validateSession(mediator, mbaasApi, excludePaths) {
```
Comment (paolobueno) [inline lib/middleware/validateSession.js:14]: need to update jsdoc to add this new param

**Comment by paolobueno** [comment]
LGTM, needs to resolve the conflicts. :+1:

**Comment by ghost** [comment]
RESOLVED. Ready for verification and merge

**Comment by paolobueno** [comment]
Closing this in favor of PR to dev branch #26

**Pull request closed without merging**